identity/handler_test.go

```diff
@@ -117,6 +117,10 @@ func TestHandler(t *testing.T) {
 		}
 	})
 
+	t.Run("case=should return 404 on a non-existing resource for identity lookup", func(t *testing.T) {
```

Comment:
Well, no. IMO, this should return an empty array, since clients expect an array here. 

How would this feature be used in the context of the ory network?
We'd probably add a "search" input to the "List identities" table, similar to the email delivery page: 
<img width="311" alt="image" src="https://user-images.githubusercontent.com/5540255/218031802-b1395f81-65d9-482e-8b8a-caefb74875d8.png">

Or do you have something else in mind?

And we already (should) have the logic for displaying a message for "no results found" if the array is empty. So if this endpoint all of a sudden returns 404 this will probably break that and we'd need to implement better error handling on that page.

Reply:
https://github.com/ory/kratos/pull/3077/commits/89565f8d071b1e7b58c9df21c6bc68fe4b084ac4